### DISCUSSION TITLE
Need help in get tree to parse to markdown

### DISCUSSION THREAD

**Discussion by BKNatani** [unifiedjs/unified#211] [Q&A]
I need to extract the headings using remark I have the tree

![image](https://user-images.githubusercontent.com/98382270/212097370-23100db8-3edb-4b69-ae65-11cd848f2e6f.png)

Not able to find any solution, Please help

**Comment by Murderlon** [comment]
Note that you provide almost no information, quality answers come from [quality questions](https://github.com/unifiedjs/.github/blob/main/support.md#asking-quality-questions). But I think you probably want https://github.com/vcarl/remark-headings to extract all the headings.

**Reply by BKNatani** [reply]
I have already check this. It is parsing from markdown file not from the delta tree

**Reply by Murderlon** [reply]
Plugins operate on the AST so you can still use it, just skip the `remark-parse` step. If you want more help, provide more contexts and code samples please.

**Reply by BKNatani** [reply]
Let me try

**Comment by ChristianMurphy** [comment]
Welcome @BKNatani!

As @Murderlon notes, your question is a bit vague.
More context would help, in particular:

* What do you want to do with the headings?
  * Could you share an example of the markdown you are working with?
  * Could you give an example of what you want to generate from the headings? 
* What version of `remark`, `react-markdown`, or `micromark` are you using?

---

Some broad guidance, `remark` and `unified` as a whole use Abstract Syntax Trees (ASTs) to work with content, and plugins are the way to access and modify those trees.
The Unified website gives an introduction to the syntax trees and plugins https://unifiedjs.com/learn/

If you are looking for inspiration on how a plugin could look. See:
* https://github.com/vcarl/remark-headings
* https://github.com/remarkjs/remark-toc
* https://github.com/mrzmmr/remark-behead
* https://github.com/Xunnamius/unified-utils/tree/main/packages/remark-capitalize-headings

Note that most leverage https://github.com/syntax-tree/unist-util-visit to find headings in the tree.

**Comment by BKNatani** [comment]
I have get it solve using https://www.npmjs.com/package/prosemirror-markdown
